docarray/utils/find.py

```diff
@@ -255,17 +260,23 @@ def _extract_embeddings(
     data: Union[AnyDocArray, BaseDoc, AnyTensor],
     search_field: str,
     embedding_type: Type,
-) -> AnyTensor:
+) -> Tuple[AnyTensor, Optional[List[int]]]:
     """Extract the embeddings from the data.
 
     :param data: the data
     :param search_field: the embedding field
     :param embedding_type: type of the embedding: torch.Tensor, numpy.ndarray etc.
-    :return: the embeddings
+    :return: a tuple of the embeddings and optionally a list of the non-null indices
     """
     emb: AnyTensor
+    valid_idx = None
     if isinstance(data, DocList):
-        emb_list = list(AnyDocArray._traverse(data, search_field))
+        emb_valid = [
+            (emb, i)
+            for i, emb in enumerate(AnyDocArray._traverse(data, search_field))
+            if emb is not None
+        ]
+        emb_list, valid_idx = zip(*emb_valid)
```

Comment:
LGTM with a minor comment: 
let's make the naming consistent. here it's called `valid_idx` and in find it's `valid_idcs`